Error building with Yandex Ads 7.0.1 on iOS

Here is the sample project https://github.com/Lerg/extension-yandexads/tree/7.0.1

It produces an error
```
duplicate interface definition for class 'AMAArrayIterator'
@interface AMAArrayIterator : NSObject <AMAResettableIterable>
```

If only `pod 'YandexMobileAds', '7.0.1'` is specified in the Podfile, another error appears
```
The abstract target Pods is not inherited by a concrete target, so the following dependencies won't make it into any targets in your project:
    - YandexMobileAds (= 7.0.1)
```

This issue on the Yandex Ads github:
https://github.com/yandexmobile/yandex-ads-sdk-ios/issues/216

> @britzl I'm sorry but I still having build errors, I've tried bob.jar (download from the alpha release on github) with the provided command. The error is different though, but there are many in the log file
> 
> [log.txt.zip](https://github.com/defold/extender/files/15440302/log.txt.zip)

At least first error is
> CocoaPodsService/Pods/KSCrash/Source/KSCrash/swift/Basic/Demangle.h:136:22: error: reference to enumeration must use 'enum' not 'enum class' [-Welaborated-enum-class]
                enum class Kind : uint16_t {

Seems like need to c++11 at least.

@britzl I've been trying different c++ versions in ext.manifest but to no avail, getting I think the same errors, here are some of many:
```
CocoaPodsService/Pods/KSCrash/Source/KSCrash/Recording/Tools/KSMachineContext.c:47:9: error: unknown type name 'ucontext64_t'; did you mean 'ucontext_t'?
typedef ucontext64_t SignalUserContext;
        ^~~~~~~~~~~~
        ucontext_t
/usr/local/extender-stage/platformsdk/iPhoneOS17.2.sdk/usr/include/sys/_types/_ucontext.h:57:33: note: 'ucontext_t' declared here
typedef _STRUCT_UCONTEXT        ucontext_t;     /* [???] user context */
                                ^
CocoaPodsService/Pods/KSCrash/Source/KSCrash/Recording/Tools/KSMachineContext.c:139:80: error: no member named 'uc_mcontext64' in 'struct __darwin_ucontext'; did you mean 'uc_mcontext'?
    _STRUCT_MCONTEXT* sourceContext = ((SignalUserContext*)signalUserContext)->UC_MCONTEXT;
                                                                               ^~~~~~~~~~~
                                                                               uc_mcontext
CocoaPodsService/Pods/KSCrash/Source/KSCrash/Recording/Tools/KSMachineContext.c:46:21: note: expanded from macro 'UC_MCONTEXT'
#define UC_MCONTEXT uc_mcontext64
                    ^
/usr/local/extender-stage/platformsdk/iPhoneOS17.2.sdk/usr/include/sys/_types/_ucontext.h:50:27: note: 'uc_mcontext' declared here
        _STRUCT_MCONTEXT        *uc_mcontext;   /* pointer to machine specific context */


CocoaPodsService/Pods/KSCrash/Source/KSCrash/swift/Basic/DemangleNodes.def:27:1: error: unknown type name 'NODE'
CONTEXT_NODE(Allocator)
^
CocoaPodsService/Pods/KSCrash/Source/KSCrash/swift/Basic/DemangleNodes.def:24:26: note: expanded from macro 'CONTEXT_NODE'
#define CONTEXT_NODE(ID) NODE(ID)
                         ^
CocoaPodsService/Pods/KSCrash/Source/KSCrash/swift/Basic/DemangleNodes.def:27:24: error: expected ';' after top level declarator
CONTEXT_NODE(Allocator)
                       ^
                       ;
2 errors generated.


CocoaPodsService/Pods/KSCrash/Source/KSCrash/swift/Basic/Demangle.h:136:22: error: reference to enumeration must use 'enum' not 'enum class' [-Welaborated-enum-class]
                enum class Kind : uint16_t {
                     ^~~~~~
CocoaPodsService/Pods/KSCrash/Source/KSCrash/swift/Basic/Demangle.h:136:44: error: expected '(' for function-style cast or type construction
                enum class Kind : uint16_t {
                                  ~~~~~~~~ ^
CocoaPodsService/Pods/KSCrash/Source/KSCrash/swift/Basic/Demangle.h:141:35: warning: alias declarations are a C++11 extension [-Wc++11-extensions]
                using IndexType = uint64_t;
                                  ^
CocoaPodsService/Pods/KSCrash/Source/KSCrash/swift/Basic/Demangle.h:148:22: warning: scoped enumerations are a C++11 extension [-Wc++11-extensions]
                enum class PayloadKind : uint8_t {
                     ^
CocoaPodsService/Pods/KSCrash/Source/KSCrash/swift/Basic/Demangle.h:148:22: error: reference to enumeration must use 'enum' not 'enum class' [-Welaborated-enum-class]
                enum class PayloadKind : uint8_t {
                     ^~~~~~
CocoaPodsService/Pods/KSCrash/Source/KSCrash/swift/Basic/Demangle.h:148:50: error: expected '(' for function-style cast or type construction
                enum class PayloadKind : uint8_t {
                                         ~~~~~~~ ^
CocoaPodsService/Pods/KSCrash/Source/KSCrash/swift/Basic/Demangle.h:154:37: error: union member 'TextPayload' has a non-trivial default constructor
                    llvm::StringRef TextPayload;
```

And here is my ext.manifest:
```
name: "YandexSdkAds"

platforms:
    common:
        context:
            flags: ["-std=c++17"]

    android:
        context:
            aaptExtraPackages: ['com.yandex.android', 'com.yandex.android.mobileads', 'com.yandex.mobile', 'com.yandex.mobile.ads', 'com.yandex.mobile.ads.imp', 'androidx.viewpager2', 'androidx.annotation', 'androidx.core', 'androidx.recyclerview', 'androidx.appcompat', 'androidx.transition', 'androidx.collection', 'androidx.appcompat.resources']

```

I'm using bob.jar downloaded from the 1.8.1-alpha release on github and using Java supplied with the Defold itself, version:
```
OpenJDK Runtime Environment Temurin-17.0.5+8 (build 17.0.5+8)
OpenJDK 64-Bit Server VM Temurin-17.0.5+8 (build 17.0.5+8, mixed mode)
```

I can't understand what are we doing different to you so it's compiling for you and not for us...

--- a/server/src/main/java/com/defold/extender/services/CocoaPodsService.java
+++ b/server/src/main/java/com/defold/extender/services/CocoaPodsService.java
@@ -1008,7 +1008,13 @@ public class CocoaPodsService {
         // https://guides.cocoapods.org/syntax/podspec.html#libraries
         spec.libraries.addAll(getAsJSONArray(specJson, "libraries"));
         if (ios != null) spec.libraries.ios.addAll(getAsJSONArray(ios, "libraries"));
+        if (spec.libraries.ios.contains("c++")) {
+            spec.flags.ios.add("-std=c++11");
+        }
         if (osx != null) spec.libraries.osx.addAll(getAsJSONArray(osx, "libraries"));
+        if (spec.libraries.osx.contains("c++")) {
+            spec.flags.osx.add("-std=c++11");
+        }
 
         // parse subspecs
         // https://guides.cocoapods.org/syntax/podspec.html#subspec

